### PR TITLE
SDK-2251 Share V2 - Create QR code

### PR DESCRIPTION
**Pull request by michaelyoti**

### CODE REVIEW

File: src/digital_identity_service/index.js [hunk 1]
```diff
@@ -80,6 +81,47 @@ class DigitalIdentityService {
...
+    const payload = new Payload({});
+
+    const requestBuilder = new RequestBuilder()
+      .withBaseUrl(this.apiUrl)
+      .withHeader('X-Yoti-Auth-Id', this.sdkId)
+      .withPemString(this.pem)
+      .withEndpoint(`/v2/sessions/${sessionId}/qr-codes`)
+      .withQueryParam('appId', this.sdkId)
+      .withMethod('POST')
+      .withPayload(payload);
```
Comment (abdalmajeed-yoti) [inline src/digital_identity_service/index.js:102]: For creating a QR code, I think there is an optional payload, should we allow the user to add them? @dhivya-pattabhiraman

Comment (dhivya-pattabhiraman) [reply src/digital_identity_service/index.js:102]: `{
  "transport": "",     // optional - default is 'QR_CODE'
  "displayMode": ""    // optional - default is 'INLINE'
}`

This one? I am not even sure what that does to be honest

File: src/digital_identity_service/index.js [hunk 2]
```diff
@@ -80,6 +81,47 @@ class DigitalIdentityService {
...
   }
+
+  /**
+   *
+   * @param sessionId
+   * @returns {Promise<ShareQrCodeCreateResult>}
+   */
+  async createQrCode(sessionId) {
+    Validation.isString(sessionId, 'sessionId');
+
```
[the remark this answers is not in the document]
Comment (michaelyoti) [reply src/digital_identity_service/index.js:92]: Yes it might be more consistent. Not sure.

File: src/digital_identity_service/index.js (hunk 1 above)
Comment (laurent-yoti) [reply src/digital_identity_service/index.js:102]: Ok, I have been investigating and had some conversations, we can ignore that bit.

File: tests/digital_identity_service/index.spec.js [hunk 3]
```diff
@@ -26,7 +26,7 @@ describe('DigitalIdentityService', () => {
   describe('#createShareSession', () => {
     const setupResponse = (responseBody, responseStatusCode = 200) => {
       nock(apiUrlDomain)
-        .post(new RegExp('/v2/sessions'))
+        .post(new RegExp('/v2/sessions[^/]'))
```
Comment (laurent-yoti) [inline tests/digital_identity_service/index.spec.js:29]: Why was that needed ?

**Review by laurent-yoti — COMMENTED** [review]
Looks good to me.

@dhivya-pattabhiraman should we add the example now ?

File: tests/digital_identity_service/index.spec.js (hunk 3 above)
Comment (michaelyoti) [reply tests/digital_identity_service/index.spec.js:29]: Because this regex also matches the URL for `createShareQrCode` which is `/v2/sessions/${sessionId}/qr-codes`. This was causing nock to give the wrong response. Even when I tried to do `nock.cleanAll()` after the tests.

Comment (laurent-yoti) [reply tests/digital_identity_service/index.spec.js:29]: Ok, I see!

**Review by laurent-yoti — APPROVED** [review]

**Review by liamd-yoti — APPROVED** [review]
LGTM

File: src/digital_identity_service/share.qr.code.create.result.js [hunk 4]
```diff
@@ -0,0 +1,41 @@
+'use strict';
+
+const Validation = require('../yoti_common/validation');
+
+/**
+ * The share QR code result
+ *
+ * @class ShareQrCodeResult
+ */
+module.exports = class ShareQrCodeResult {
```
Comment (dhivya-pattabhiraman) [inline src/digital_identity_service/share.qr.code.create.result.js]: Shouldn't this be ShareQrCodeCreateResult ?

**Review by dhivya-pattabhiraman — APPROVED** [review]

File: src/digital_identity_service/share.qr.code.create.result.js (hunk 4 above)
Comment (laurent-yoti) [reply src/digital_identity_service/share.qr.code.create.result.js]: 👍
